test/Microsoft.TestPlatform.CommunicationUtilities.UnitTests/TestRequestSenderTests.cs

```diff
@@ -470,6 +470,18 @@ public void DiscoverTestShouldNotifyLogMessageIfClientDisconnectedWithClientExit
         _mockDiscoveryEventsHandler.Verify(eh => eh.HandleRawMessage(It.Is<string>(s => !string.IsNullOrEmpty(s) && s.Equals("Serialized Stderr"))), Times.Once);
     }
 
+    [TestMethod]
+    public void DiscoverTestShouldNotifyDiscoveryCompleteIfClientDisconnectedBeforeDiscovery()
+    {
+        SetupFakeCommunicationChannel();
+
+        RaiseClientDisconnectedEvent();
+
+        _testRequestSender.DiscoverTests(new DiscoveryCriteria(), _mockDiscoveryEventsHandler.Object);
```

Comment:
Could we add tests for the other methods that are also providing this new behavior?

Reply:
I have added the additional tests